libtiny_client/src/state.rs

```diff
@@ -323,7 +333,7 @@ impl StateInner {
 
                 match parse_servername(params) {
                     None => {
-                        // TODO: Log
+                        error!("Could not parse server name in 002 RPL_YOURHOST message.");
```

Comment:
This case is more interesting, and I think I found a potential bug while looking at uses of `self.servername`. Currently we use the server name when sending pings, and we don't send pings if we're unable to parse the server name from RPL_YOURHOST, but sending pings is pretty important as that's how we check connectivity. I think if we're unable to parse server name we can use the server address as the ping target. Or maybe just send a PING without an argument and wait for an error message from the server (which should reset the ping timer). For example, this works fine to check connectivity:

```
>>> PING <-- sent message
13:33 cherryh.freenode.net: osa1 PING Not enough parameters
```

When we receive this error message the pinger timer will be reset.

Only concern is if we can't parse a server name and send PING messages without a target then the server tab will have lots of "PING not enough parameters" lines. Maybe that's fine though as this should be a rare case.

For this PR I think it's fine to add an error-level log here.